File: app/src/main/java/com/awave/apps/droider/Main/NotifyService.java
package com.awave.apps.droider.Main;

import android.app.Notification;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.res.Resources;
import android.graphics.BitmapFactory;
import android.os.Build;
import android.os.IBinder;
import android.preference.PreferenceManager;
import android.util.Log;

import com.awave.apps.droider.R;

import java.util.Timer;
import java.util.TimerTask;

public class NotifyService extends Service {
    private static final int NOTIFY_ID = 101;
    private Notification notification;
    private SharedPreferences sp;
    private Timer myTimer = new Timer();
    private NotificationManager notificationManager;

    @Override
    public void onCreate() {
        sp = PreferenceManager.getDefaultSharedPreferences(this);
        super.onCreate();
        notificationManager = (NotificationManager) getApplicationContext()
                .getSystemService(Context.NOTIFICATION_SERVICE);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (isPushEnabled()) {
            try {
                myTimer.schedule(new TimerTask() {
                    @Override
                    public void run() {
                        Notify();
                    }
                }, NewPush(Interval()), NewPush(Interval()));
                Log.d("NOTIFY", "Repeat in " + Interval() + "hour");
            } catch (IllegalArgumentException e) {
                e.printStackTrace();
            }
        }

        return START_REDELIVER_INTENT;
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    private byte Interval() {
        if (Byte.parseByte(sp.getString("interval", "3")) == 3)
            return 3;
        else if (Byte.parseByte(sp.getString("interval", "6")) == 6)
            return 6;
        else if (Byte.parseByte(sp.getString("interval", "12")) == 12)
            return 12;
        else if (Byte.parseByte(sp.getString("interval", "24")) == 24)
            return 24;
        else
            return 1;
    }

    private boolean isPushEnabled() {
        return sp.getBoolean("notify", false);
    }

    private long NewPush(long time) {
        return (time * 1000 * 60 * 60);
    }  //time * 1000 * 60 * 60 = time часов

    private void Notify() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                Intent notifyIntent = new Intent(getApplicationContext(), MainActivity.class);
                PendingIntent contextIntent = PendingIntent.getActivity(getApplicationContext(), 0, notifyIntent, PendingIntent.FLAG_CANCEL_CURRENT);

                Resources res = getApplicationContext().getResources();
                Notification.Builder builder = new Notification.Builder(getApplicationContext());

                builder.setContentIntent(contextIntent)
                        .setSmallIcon(R.drawable.ic_stat_dr)
                        .setLargeIcon(BitmapFactory.decodeResource(res, R.mipmap.ic_large_notify))
                        .setTicker("Droider")
                        .setWhen(System.currentTimeMillis())
                        .setAutoCancel(true)
                        .setContentTitle("Новые статьи")
                        .setContentText("Проверь, что нового?");

                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
                    notification = builder.getNotification();
                } else {
                    Notification.InboxStyle inboxStyle = new Notification.InboxStyle();

                    inboxStyle.setBigContentTitle("");
                    inboxStyle.addLine("Привет, не хочешь ли узнать, что нового");
                    inboxStyle.addLine(" в этом технологическом мире?");


                    builder.setStyle(inboxStyle);
                    notification = builder.build();
                }
                notification.defaults = Notification.DEFAULT_VIBRATE | Notification.DEFAULT_LIGHTS;

                Log.d("Notify ", "Сработало");
                notificationManager.notify(NOTIFY_ID, notification);
            }
        }).start();
    }
}

File: app/src/main/java/com/apps/wow/droider/NavDrawScreens/NotifyService.kt
package com.apps.wow.droider.NavDrawScreens

import android.app.Notification
import android.app.NotificationManager
import android.app.PendingIntent
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.IBinder
import android.preference.PreferenceManager
import android.support.v4.app.NotificationCompat
import android.util.Log
import com.apps.wow.droider.Feed.FeedActivity
import com.apps.wow.droider.Feed.Interactors.FeedLoadingInteractor
import com.apps.wow.droider.R
import com.apps.wow.droider.Utils.Const
import java.util.*


class NotifyService : Service() {
    private lateinit var mSharedPrefs: SharedPreferences
    private val mTimer = Timer()
    val mFeedLoadingInteractor = FeedLoadingInteractor()

    private var mPostTitle: String? = null


    override fun onCreate() {
        super.onCreate()
        mSharedPrefs = PreferenceManager.getDefaultSharedPreferences(this)
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        if (isPushEnabled) {
            try {
                mTimer.schedule(object : TimerTask() {
                    override fun run() {
                        loadLastPost()
                    }
                }, newPush(interval().toLong()), newPush(interval().toLong()))
                Log.d("NOTIFY", "Repeat in " + interval() + "hour")
            } catch (e: IllegalArgumentException) {
                e.printStackTrace()
            }

        } else {
            stopSelf()
        }

        return Service.START_REDELIVER_INTENT
    }

    private fun loadLastPost() {
        mFeedLoadingInteractor.loadFeed(Const.CATEGORY_MAIN, Const.SLUG_MAIN, 1, 0).subscribe({
            if (it.isSuccessful && !it.body()?.posts?.isEmpty()!!) {
                if (mSharedPrefs.getString(Const.PREF_NAME_URL, null) != it.body()?.posts?.get(0)?.url) {
                    mSharedPrefs.edit().putString(Const.PREF_NAME_URL, it.body()?.posts?.get(0)?.url).apply()
                    mPostTitle = it.body()?.posts?.get(0)?.titleValue
                    buildAndSendNotification()
                }
            }
        })
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    private fun interval(): Byte {
        if (mSharedPrefs.getString("interval", "3").toInt() == 3)
            return 3
        else if (mSharedPrefs.getString("interval", "6").toInt() == 6)
            return 6
        else if (mSharedPrefs.getString("interval", "12").toInt() == 12)
            return 12
        else if (mSharedPrefs.getString("interval", "24").toInt() == 24)
            return 24
        else
            return 1
    }

    private val isPushEnabled: Boolean
        get() = mSharedPrefs.getBoolean("notify", false)

    //time * 1000 * 60 * 60 = time часов
    private fun newPush(time: Long): Long {
        return time * 1000 * 60 * 60
    }

    private fun buildAndSendNotification() {

        val channelId = "com.apps.wow.droider"
        val builder = NotificationCompat.Builder(applicationContext, channelId)

        val notifyIntent = Intent(applicationContext, FeedActivity::class.java)
        val contextIntent = PendingIntent.getActivity(
                applicationContext, 0, notifyIntent,
                PendingIntent.FLAG_CANCEL_CURRENT)

        builder.setContentIntent(contextIntent)
                .setSmallIcon(R.mipmap.ic_stat_notify_article)
                .setTicker(getString(R.string.app_name))
                .setWhen(System.currentTimeMillis())
                .setAutoCancel(true)
                .setContentTitle(getString(R.string.notification_title))
                .setContentText(mPostTitle)
                .setDefaults(Notification.DEFAULT_VIBRATE or Notification.DEFAULT_LIGHTS)

        val notification = builder.build()

        (getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager).notify(Const.NOTIFY_ID, notification)
    }
}
